pkg/webhook/admission/servingruntime/servingruntime_webhook.go

```diff
@@ -0,0 +1,176 @@
+		if err := validateServingRuntimePriority(&servingRuntime.Spec, &runtime.Spec, servingRuntime.Name, runtime.Name); err != nil {
+			return admission.Denied(fmt.Sprintf(InvalidPriorityServingRuntimeError, err.Error(), runtime.Name, servingRuntime.Name, servingRuntime.Namespace))
+		}
+	}
+	return admission.Allowed("")
+}
+
+// Handle validates the incoming request
+func (csr *ClusterServingRuntimeValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	clusterServingRuntime := &v1alpha1.ClusterServingRuntime{}
+	if err := csr.Decoder.Decode(req, clusterServingRuntime); err != nil {
+		log.Error(err, "Failed to decode cluster serving runtime", "name", clusterServingRuntime.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	ExistingRuntimes := &v1alpha1.ClusterServingRuntimeList{}
+	if err := csr.Client.List(context.TODO(), ExistingRuntimes); err != nil {
+		log.Error(err, "Failed to get cluster serving runtime list")
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// Only validate for priority if the new cluster serving runtime is not disabled
+	if clusterServingRuntime.Spec.IsDisabled() {
+		return admission.Allowed("")
+	}
+
+	for _, runtime := range ExistingRuntimes.Items {
+		if err := validateServingRuntimePriority(&clusterServingRuntime.Spec, &runtime.Spec, clusterServingRuntime.Name, runtime.Name); err != nil {
+			return admission.Denied(fmt.Sprintf(InvalidPriorityClusterServingRuntimeError, err.Error(), runtime.Name, clusterServingRuntime.Name))
+		}
+	}
+	return admission.Allowed("")
+}
+
+func validateServingRuntimePriority(newSpec *v1alpha1.ServingRuntimeSpec, existingSpec *v1alpha1.ServingRuntimeSpec, existingRuntimeName string, newRuntimeName string) error {
+	// Skip the runtime if it is disabled or both are not multi model runtime or the stale runtime in the api server
+	if (newSpec.IsMultiModelRuntime() != existingSpec.IsMultiModelRuntime()) || (existingSpec.IsDisabled()) || (existingRuntimeName == newRuntimeName) {
+		return nil
+	}
+	// Only validate for priority if both servingruntimes supports the same protocol version
+	isTheProtocolSame := false
+	for _, protocolVersion := range existingSpec.ProtocolVersions {
+		if contains(newSpec.ProtocolVersions, protocolVersion) {
+			isTheProtocolSame = true
+			break
+		}
+	}
+	if isTheProtocolSame {
+		for _, existingModelFormat := range existingSpec.SupportedModelFormats {
+			for _, newModelFormat := range newSpec.SupportedModelFormats {
+				// Only validate priority if autoselect is ture
+				if (strings.EqualFold(existingModelFormat.Name, newModelFormat.Name) && existingModelFormat.IsAutoSelectEnabled() && newModelFormat.IsAutoSelectEnabled()) &&
+					((existingModelFormat.Version == nil && newModelFormat.Version == nil) ||
+						(existingModelFormat.Version != nil && newModelFormat.Version != nil && *existingModelFormat.Version == *newModelFormat.Version)) {
```

Comment:
I think it is better to extract to a separate function to check whether the two serving runtime supports the same model format and version.

Reply:
Done